test/integration/etcd_storage_path_test.go

```diff
@@ -129,24 +130,77 @@ var etcdStorageData = map[reflect.Type]struct {
-	reflect.TypeOf(&authorizationapiv1.LocalResourceAccessReview{}):     {ephemeral: true}, // TODO(mo): Just making the test pass
-	reflect.TypeOf(&authorizationapiv1.Role{}):                          {ephemeral: true}, // TODO(mo): Just making the test pass
-	reflect.TypeOf(&authorizationapiv1.RoleBindingRestriction{}):        {ephemeral: true}, // TODO(mo): Just making the test pass
-	reflect.TypeOf(&authorizationapiv1.Policy{}):                        {ephemeral: true}, // TODO(mo): Just making the test pass
-	reflect.TypeOf(&authorizationapiv1.ClusterPolicyBinding{}):          {ephemeral: true}, // TODO(mo): Just making the test pass
-	reflect.TypeOf(&authorizationapiv1.SubjectAccessReviewResponse{}):   {ephemeral: true}, // TODO(mo): Just making the test pass
-	reflect.TypeOf(&authorizationapiv1.ClusterRole{}):                   {ephemeral: true}, // TODO(mo): Just making the test pass
-	reflect.TypeOf(&authorizationapiv1.SubjectRulesReview{}):            {ephemeral: true}, // TODO(mo): Just making the test pass
+	reflect.TypeOf(&authorizationapiv1.PolicyBinding{}): {
+		stub: &authorizationapiv1.PolicyBinding{
+			RoleBindings: authorizationapiv1.NamedRoleBindings{
+				{
+					Name: "rb",
+					RoleBinding: authorizationapiv1.RoleBinding{
+						ObjectMeta: kapiv1.ObjectMeta{Name: "rb", Namespace: testNamespace},
+						RoleRef:    kapiv1.ObjectReference{Name: "r"},
+					},
+				},
+			},
+		},
+		expectedEtcdPath: "openshift.io/authorization/local/policybindings/etcdstoragepathtestnamespace/:default",
+	},
+	reflect.TypeOf(&authorizationapiv1.ClusterPolicyBinding{}): {
+		stub: &authorizationapiv1.ClusterPolicyBinding{
+			ObjectMeta: kapiv1.ObjectMeta{Name: "objectisincomparewhitelist"},
+		},
+		expectedEtcdPath: "openshift.io/authorization/cluster/policybindings/:default",
+	},
+	reflect.TypeOf(&authorizationapiv1.Policy{}): {
+		stub: &authorizationapiv1.Policy{
+			Roles: authorizationapiv1.NamedRoles{
+				{
+					Name: "r",
+					Role: authorizationapiv1.Role{
+						ObjectMeta: kapiv1.ObjectMeta{Name: "r", Namespace: testNamespace},
+					},
+				},
+			},
+		},
+		expectedEtcdPath: "openshift.io/authorization/local/policies/etcdstoragepathtestnamespace/default",
+	},
+	reflect.TypeOf(&authorizationapiv1.ClusterPolicy{}): {
+		stub: &authorizationapiv1.ClusterPolicy{
+			ObjectMeta: kapiv1.ObjectMeta{Name: "objectisincomparewhitelist"},
+		},
+		expectedEtcdPath: "openshift.io/authorization/cluster/policies/default",
+	},
+	reflect.TypeOf(&authorizationapiv1.RoleBindingRestriction{}): {
+		stub: &authorizationapiv1.RoleBindingRestriction{
+			ObjectMeta: kapiv1.ObjectMeta{Name: "rbr"},
+			Spec: authorizationapiv1.RoleBindingRestrictionSpec{
+				ServiceAccountRestriction: &authorizationapiv1.ServiceAccountRestriction{
+					ServiceAccounts: []authorizationapiv1.ServiceAccountReference{
+						{
+							Name: "sa",
+						},
+					},
+				},
+			},
+		},
+		expectedEtcdPath: "openshift.io/rolebindingrestrictions/rbr",
```

Comment:
`oc create -f` in `myproject`

```yaml
apiVersion: v1
kind: RoleBindingRestriction
metadata:
  name: rbr
spec:
  serviceaccountrestriction:
    serviceaccounts:
    - name: sa
```

```
I0125 22:51:03.105611   21605 round_trippers.go:299] curl -k -v -XGET  -H "Accept: application/json" -H "User-Agent: oc/v1.5.2+43a9be4 (linux/amd64) kubernetes/43a9be4" https://10.13.129.52:8443/oapi/v1/namespaces/default/rolebindingrestrictions
I0125 22:51:03.137310   21605 round_trippers.go:318] GET https://10.13.129.52:8443/oapi/v1/namespaces/default/rolebindingrestrictions 200 OK in 31 milliseconds
I0125 22:51:03.137390   21605 round_trippers.go:324] Response Headers:
I0125 22:51:03.137414   21605 round_trippers.go:327]     Cache-Control: no-store
I0125 22:51:03.137428   21605 round_trippers.go:327]     Content-Type: application/json
I0125 22:51:03.137446   21605 round_trippers.go:327]     Content-Length: 558
I0125 22:51:03.137456   21605 round_trippers.go:327]     Date: Thu, 26 Jan 2017 03:51:03 GMT
I0125 22:51:03.137632   21605 request.go:904] Response Body: {"kind":"RoleBindingRestrictionList","apiVersion":"v1","metadata":{"selfLink":"/oapi/v1/namespaces/default/rolebindingrestrictions","resourceVersion":"1084"},"items":[{"metadata":{"name":"rbr","namespace":"myproject","selfLink":"/oapi/v1/namespaces/myproject/rolebindingrestrictions/rbr","uid":"6ddb2bdd-e37a-11e6-bf25-507b9dac97ff","resourceVersion":"1064","creationTimestamp":"2017-01-26T03:49:24Z"},"spec":{"userrestriction":null,"grouprestriction":null,"serviceaccountrestriction":{"serviceaccounts":[{"name":"sa","namespace":""}],"namespaces":null}}}]}
I0125 22:51:03.138707   21605 cached_discovery.go:112] returning cached discovery info from /home/mkhan/.kube/10.13.129.52_8443/servergroups.json
F0125 22:51:03.139297   21605 helpers.go:116] the namespace from the provided object "myproject" does not match the namespace "default". You must pass '--namespace=myproject' to perform this operation.
```

Something does not seem right:

Get 10.13.129.52:8443/oapi/v1/namespaces/**default**/rolebindingrestrictions

Response:
"selfLink":"/oapi/v1/namespaces/**default**/rolebindingrestrictions"
"items": "selfLink":"/oapi/v1/namespaces/**myproject**/rolebindingrestrictions/rbr"